hoverboard_joint_publisher:
  type: "joint_state_controller/JointStateController"
  publish_rate: 50

hoverboard_velocity_controller:
  type        : "diff_drive_controller/DiffDriveController"
  left_wheel  : ['front_left_wheel_joint', 'rear_left_wheel_joint']
  right_wheel : ['front_right_wheel_joint', 'rear_right_wheel_joint']
  publish_rate: 50.0 # default: 50
  pose_covariance_diagonal : [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
  twist_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]

  # Wheel separation and diameter.
  wheel_separation : 0.472
  # Measured circumference with a ruler: (0.543 m / pi) / 2
  wheel_radius : 0.086421134

  # Wheel separation and radius multipliers
  wheel_separation_multiplier: 1.0 # default: 1.0
  wheel_radius_multiplier    : 1.0 # default: 1.0

  # Velocity commands timeout [s], default 0.5
  cmd_vel_timeout: 0.3

  # Base frame_id
  base_frame_id: base_footprint
  allow_multiple_cmd_vel_publishers: true
  odom_frame_id: '/odom'

  # Odometry fused with IMU is published by robot_localization, so
  # no need to publish a TF based on encoders alone.
  enable_odom_tf: true

  # Velocity and acceleration limits
  # Whenever a min_* is unspecified, default to -max_*
  linear:
    x:
      has_velocity_limits    : true # was true
      max_velocity           : 1.0  # m/s
      has_acceleration_limits: true # was true
      max_acceleration       : 1.0  # m/s^2
      has_jerk_limits        : false # was true
      max_jerk               : 1.0  # m/s^3
  angular:
    z:
      has_velocity_limits    : true # was true
      max_velocity           : 6.28 # rad/s
      has_acceleration_limits: true # was true
      max_acceleration       : 3.14 # rad/s^2
      has_jerk_limits        : false # was true
      max_jerk               : 3.14 # rad/s^3
